### DISCUSSION TITLE
New Centos 9 and GA Network Install Fails - Packaging Errors

### DISCUSSION THREAD

**Discussion by trb143** [Security-Onion-Solutions/securityonion#11107] [2.4]
### Version

2.4.10

### Installation Method

Network installation on Red Hat derivative like Oracle, Rocky, Alma, etc.

### Description

installation

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Meets minimum requirements

### Network Traffic Collection

span port

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

The Install run and failed and accessing the logs the following has been noted.

1) Tried to install MariaDB-devel and failed.  Should be mariadb-devel.
              errors:
                  - Running scope as unit: run-r3f065ce103304a9aac08240af9ca6a34.scope
                    Last metadata expiration check: 0:00:01 ago on Fri Aug 18 16:06:18 2023.
                    No match for argument: MariaDB-devel
                      * Maybe you meant: mariadb-devel

Checking packages.sls there is a test for Centos which looks incorrect.  This failure has other issues the following were not installed.

Python3-rich , python3-docker, python3-dnf-plugin-versionlock, python3-packaging, python3-watchdog, sqlite.

The result was authentication errors between containers.
so-elastalert, so-elasticfleet and so-soc failed to start and were missing after server restart.

2)

Other errors were spotted but may be connected to above.

--
                  mode:
                      0644
                  user:
                      930
              /opt/so/conf/elasticsearch/templates/component/ecs/error.json:
                  ----------
                  diff:
                      New file
                  group:
--
                  Waiting for value 'Elastic' at 'http://localhost:5601/app/kibana' (14/300)
                  Server is not ready
                  Waiting for value 'Elastic' at 'http://localhost:5601/app/kibana' (15/300)
                  Received expected response; proceeding.
                  {"statusCode":415,"error":"Unsupported Media Type","message":"Unsupported Media Type"}
----------
          ID: securitySolution_saved_objects
    Function: file.managed
        Name: /opt/so/conf/kibana/securitySolution_saved_objects.ndjson.template
--
                  100 5748k  100   126  100 5748k  15750   701M --:--:-- --:--:-- --:--:--  701M
              stdout:
                  Waiting for value 'Elastic' at 'http://localhost:5601/app/kibana' (1/300)
                  Received expected response; proceeding.
                  {"statusCode":413,"error":"Request Entity Too Large","message":"Payload content length greater than maximum allowed: 1048576"}

### Create ES Token ###
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    30  100    30    0     0  15000      0 --:--:-- --:--:-- --:--:-- 15000
parse error: Invalid numeric literal at line 1, column 7

Add Manager Elasticsearch Output...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
--
100   191  100    30  100   161  15000  80500 --:--:-- --:--:-- --:--:-- 95500
Kibana server is not ready yet

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by dougburks** [comment]
We'll take a look at this as time allows. In the meantime, have you tried using our Security Onion ISO image?
https://docs.securityonion.net/en/2.4/os.html

**Comment by trb143** [comment]
I think I will try that next as there seem to be other issues.  There is a second discission raised with Centos.
